bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/DecimalValue.java

```diff
@@ -253,7 +253,7 @@ public Object frozenCopy(Map<Object, Object> refs) {
      * @param parent The link to the parent node
      */
     public String stringValue(BLink parent) {
-        if (this.valueKind != DecimalValueKind.OTHER) {
+        if (this.valueKind != DecimalValueKind.OTHER && this.valueKind != DecimalValueKind.ZERO) {
```

Comment:
Consider these three cases,
**Case 1**
```ballerina
public function main() {
   decimal d1 = 0.0;
   decimal d2 = 0E-1;
   assertEquality(d1.toString(), d2.toString()); // true
}
```
**Case 2**

```ballerina
public function main() {
   decimal d1 = 1000;
   decimal d2 = 1E3;
   assertEquality(d1.toString(), d2.toString()); // false
}
```
**Case 3**
```ballerina
public function main() {
   decimal d1 = 0.1;
   decimal d2 = 1E-1;
   assertEquality(d1.toString(), d2.toString()); // true
}
```
Is the output of first and third cases correct?  According to the spec result of `ToString()` should be in the expression style(not considering the value). Isn't it? @MaryamZi @KavinduZoysa

Reply:
As per the offline discussion created separate issue to address this #34648